spec/formatting.md

```diff
@@ -151,12 +154,16 @@ the following steps are taken:
    and use a _fallback value_ for the _expression_.
 3. Resolve the _option_ values to a mapping of string identifiers to values.
    For each _option_:
-     * If its right-hand side successfully resolves to a value,
-       bind the _name_ of the _option_ to the resolved value in the mapping.
-     * Otherwise, do not bind the _name_ of the _option_ to any value in the mapping.
-4. Call the function implementation with the following arguments:
-
-   - The current _locale_.
+   - If its right-hand side successfully resolves to a value,
+     bind the _name_ of the _option_ to the resolved value in the mapping.
+   - Otherwise, do not bind the _name_ of the _option_ to any value in the mapping.
+4. Determine the _expression_ _locale_.
+   If the _expression_ has a valid `@locale` _expression attribute_,
+   use its resolved value as the _expression_ _locale_.
+   Otherwise, look up the _message_ _locale_ from the _formatting context_.
+5. Call the function implementation with the following arguments:
+
+   - The _expression_ _locale_.
```

Comment:
I would expose the expression attributes here also. Locale may not be the only one--I certainly have timezone in mind and other attributes may be provided by the implementation or our specification that affect the formatter.

Reply:
At least for the JS `Intl.DateTimeFormat`, `timeZone` is a regular option. Should we not handle it as such?

For other attributes, there is this below:
> Implementations MAY assign meaning during formatting to other _expression attributes_

Should the text here make a reference to that as well, or is that implicit permission sufficiently clear?

Reply:
> For other attributes, there is this below:
>> Implementations MAY assign meaning during formatting to other expression attributes
> Should the text here make a reference to that as well, or is that implicit permission sufficiently clear?

You're permitting _implementations_ to assign meaning, but here I'm suggesting that _functions_ can assign meaning to attribute values, including those unknown to the implementation. They can only do this if they have access to the values, of course.